madminer/utils/morphing.py

```diff
@@ -123,41 +129,117 @@ def find_components(self, max_overall_power=4):
             The maximal sum of powers of all parameters contributing to the squared matrix element.
             Typically, if parameters can affect the couplings at n vertices, this number is 2n. Default value: 4.
 
+        BSM_max_power : int, optional
+            The maximal sum of powers of all parameters contributing to the squared matrix element for the BSM couplings.
+
+        Nd : int, optional
+            The number of decay vertices.
+
+        Np : int, optional
+            The number of production vertices.
+
+        Ns : int, optional
+            The number of couplings that works both for production and decay.
+
         Returns
         -------
         components : ndarray
             Array with shape (n_components, n_parameters), where each entry gives the power with which a parameter
             scales a given component.
         """
+        if Nd != 0 or Np != 0 or Ns != 0:
+            lst = []
```

Comment:
Could you be more precise here? I am not sure what this _list_ is suppose to contain by its name.

Reply:
Thank you for your feedbacks, i'll change it name to components in the next commit.